src/main/java/ejercicio/banco/repository/CsvBankRepository.java

```diff
@@ -0,0 +1,105 @@
+package main.java.ejercicio.banco.repository;
+
+import main.java.ejercicio.banco.dto.Bank;
+import main.java.ejercicio.banco.util.CsvReader;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CsvBankRepository implements BankRepository {
+
+    private CsvReader reader;
+
+    public CsvBankRepository(){
+        this.reader = new CsvReader();
+    }
+
+    @Override
+    public List<Bank> findAll(String filename) throws FileNotFoundException {
+        // Creates list of Banks
+        List<Bank> banks = new ArrayList<Bank>();
+        List<String> in = reader.read(filename);
+
+        for (int x = 0; x<in.size(); x++) {
+            String line = in.get(x);
+            System.out.println(line);
+
+            // Assuming that the info is separated by "; " ,splits it
+            String[] parts = line.split(";");
+            int bankId = Integer.parseInt(parts[0]);
+            String bankName = parts[1];
+            String bankAddress = parts[2];
+
+            //Prints bank info in concolse
+            System.out.println("Our bank "+bankName+" has been processed");
+
+
+            // Creates a new bank
+            Bank bnk = new Bank(bankId, bankName, bankAddress);
+            banks.add(bnk);
+        }
+        return banks;
+    }
+
+    @Override
+    public Bank find(int id) throws FileNotFoundException {
```

Comment:
What do you think if we implement the method `find` by using the `findAll` to get all the processed banks and after we only need to get the correct id?

I mean something like:


```
@Override
public Bank find(int id) throws FileNotFoundException {
   List<Bank> all = findAll(FILENAME);

   for (Bank bank : all) {
      if (id == bank.getId) {
         return bank;
      }
   }
}
```